datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1067,18 +1096,40 @@ impl SMJStream {
                 .collect::<Result<Vec<_>, ArrowError>>()
                 .map_err(Into::<DataFusionError>::into)?;
 
+            // Create null streamed (left) columns
             let mut streamed_columns = self
                 .streamed_schema
                 .fields()
                 .iter()
                 .map(|f| new_null_array(f.data_type(), buffered_indices.len()))
                 .collect::<Vec<_>>();
 
+            let filter_columns =
```

Comment:
Inside `get_filter_column`, the filter is checked if it is `Some`.  If not, the result will be an empty vector.

Instead of doing that, you can move the filter columns calculation under 
`let output_batch = if let Some(f) = &self.filter {` and make `get_filter_column` expects `joinfilter_filter: &JoinFilter`.

Reply:
It is because `buffered_columns` is consumed by `streamed_columns.extend(buffered_columns);`. So either I clone `buffered_columns` before it, or put `get_filter_column ` before it like I current do.